File: Excel/900-999/901/901 Challenge.R
library(tidyverse)
library(readxl)

path <- "Excel/900-999/901/901 First 10 Odd Numbers.xlsx"
test <- read_excel(path, range = "A1:A101") %>% pull() %>% as.numeric()

res <- seq(101, 1e7, 2) |>
  tibble(n = _) |>
  mutate(
    f = as.integer(str_sub(n, 1, 1)),
    l = as.integer(str_sub(n, -1))
  ) |>
  filter(
    f != l,
    f != 0,
    l != 0,
    paste0(f, l) != "01",
    n %% as.integer(paste0(f, l)) == 0,
    n %% as.integer(paste0(l, f)) == 0
  ) |>
  slice_head(n = 100) |>
  pull(n)

all.equal(res, test)
#[1] TRUE

File: Excel/900-999/901/901 Challenge.py
import pandas as pd

path = "Excel\\900-999\\901\\901 First 10 Odd Numbers.xlsx"
test = pd.read_excel(path, usecols='A', nrows = 101)

res = []
n = 101

while len(res) < 100:
    if n % 2:
        s = str(n)
        f, l = int(s[0]), int(s[-1])

        if f != l and f != 0 and l != 0 and s[0] + s[-1] != "01":
            a = int(s[0] + s[-1])
            b = int(s[-1] + s[0])

            if n % a == 0 and n % b == 0:
                res.append(n)
    n += 2

print(res == test['Answer Expected'].tolist()) # True